contracts/andromeda_cw721/src/contract.rs

```diff
@@ -0,0 +1,344 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    has_coins, to_binary, BankMsg, Binary, Coin, CosmosMsg, Deps, DepsMut, Empty, Env, MessageInfo,
+    Reply, Response, StdError, Storage, SubMsg,
+};
+
+use andromeda_protocol::{
+    communication::{
+        hooks::AndromedaHook,
+        modules::{
+            execute_alter_module, execute_deregister_module, execute_register_module, module_hook,
+            on_funds_transfer, validate_modules, ADOType, MODULE_ADDR, MODULE_INFO,
+        },
+        parse_message, AndromedaMsg,
+    },
+    cw721::{ExecuteMsg, InstantiateMsg, QueryMsg, TokenExtension, TransferAgreement},
+    error::ContractError,
+    operators::execute_update_operators,
+    ownership::{execute_update_owner, CONTRACT_OWNER},
+    rates::Funds,
+    require,
+    response::get_reply_address,
+};
+use cw721_base::{state::TokenInfo, Cw721Contract};
+
+pub type AndrCW721Contract<'a> = Cw721Contract<'a, TokenExtension, Empty>;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    CONTRACT_OWNER.save(deps.storage, &info.sender)?;
+
+    let sender = info.sender.as_str();
+    let mut resp = Response::default();
+    if let Some(modules) = msg.modules.clone() {
+        validate_modules(&modules, ADOType::CW721)?;
+        for module in modules {
+            resp = execute_register_module(
```

Comment:
I think we need to do something like 
```
let response = execute_register_module(..);
resp = resp.add_submessages(response.msgs);
```
I made the same error before and it resulted in only the last module being instantiated.